packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+import {
+  CommittedTransactionInfo,
+  GatewayApiClient,
+  GatewayStatusResponse,
+  StateEntityDetailsRequest,
+  StateEntityDetailsResponse,
+  TransactionCommittedDetailsRequest,
+  TransactionCommittedDetailsResponse,
+  TransactionPreviewRequest,
+  TransactionPreviewResponse,
+  TransactionSubmitResponse,
+} from '@radixdlt/babylon-gateway-api-sdk'
+import {
+  Convert,
+  Curve,
+  LTSRadixEngineToolkit,
+  NetworkId,
+  PrivateKey,
+  RadixEngineToolkit,
+  TransactionBuilder,
+  TransactionManifest,
+  generateRandomNonce,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeOption,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxHistoryParams,
+  TxParams,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto/lib'
+import { Address, Asset, assetAmount, assetToBase, baseAmount } from '@xchainjs/xchain-util'
+import BIP32Factory, { BIP32Interface } from 'bip32'
+import { derivePath } from 'ed25519-hd-key'
+import * as ecc from 'tiny-secp256k1'
+// eslint-disable-next-line ordered-imports/ordered-imports
+import {
+  AssetXRD,
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STOKENET_GATEWAY_URL,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+  xrdRootDerivationPaths,
+} from './const'
+
+/**
+ * Custom Radix client
+ */
+
+export default class Client extends BaseXChainClient {
+  gatewayApiClient: GatewayApiClient
+  curve: Curve
+  constructor(params: XChainClientParams, curve: Curve) {
+    super(RadixChain, {
+      network: params.network,
+      phrase: params.phrase,
+      rootDerivationPaths: xrdRootDerivationPaths,
+      feeBounds: params.feeBounds,
+    })
+    this.curve = curve
+    this.gatewayApiClient = GatewayApiClient.initialize({
+      networkId: this.getRadixNetwork(),
+      applicationName: 'xchainjs',
+    })
+  }
+
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
```

Comment:
Discussion: I think that we need to account for messages in the preview to get an accurate estimate.

Reply:
Max plaintext message is 2048 so it might be a good idea to include a max length message in preview so that we get more accurate fee estimate.